### PR TITLE
Changes to Logging and Initialization of the CLM Lake Model

### LINKED ISSUE #93: CLM Lake initialization issues cause an energy budget problem
# Description

Some 3D lake variables are either initialized only on one 2D slice, or have the same values in all levels of a column. This unphysical situation causes energy budget issues in the CLM Lake Model.

@tanyasmirnova discovered this problem

## Steps to Reproduce

Run the model with clm_lake_debug turned on and you'll see troubles reported for the first hour or so. Some of the surface fields will lack lake ice for the first hour or so, as a consequence of this.

## Additional Context

* Machine - hera
* Compiler - intel
* Suite Definition File or Scheme - FV3_HRRR

## Output

No.

### PR DESCRIPTION
**Pull request by SamuelTrahanNOAA**
This is the work of @tanyasmirnova to fix some initialization issues in the CLM Lake Model. Initialization appears to work correctly without fractional ice. There are some lingering issues with fractional ice initialization that we're working on.

- fixes #93 

Changes:

1. Use ice thickness hice(i) to find the level in the lake where ice is zero.
2. Do not allow lake temperature to be below freezing point if there is no ice.
3. If there is no snow or ice, do not allow surface lake temperature to be below freezing point. These changes fixed the problem with large errors in the energy budget at the beginning of the cold-start run with lakes.
4. Added flag to turn on debug print statements in the CLM lake model.

### CODE REVIEW

**Comment by SamuelTrahanNOAA** [comment]
This wasn't the final version of Tanya's changes. I'm retesting her final version now, on top of the latest ufs-weather-model.

**Review by tanyasmirnova — APPROVED** [review]
@SamuelTrahanNOAA Thank you, Sam, for making a PR with these modification to the CLM lake initialization.

**Review by grantfirl — APPROVED** [review]
I don't see any issues from a CCPP point of view. There is a lot more debugging output, of course, but it looks to be controlled by debug flags, so it shouldn't be an issue.

**Comment by SamuelTrahanNOAA** [comment]
> There is a lot more debugging output, of course, but it looks to be controlled by debug flags, so it shouldn't be an issue.

The amount of output from clm_lake_debug=.true. was so large that it was useless. Tanya split it up and added a separate flag to enable the super-verbose prints.

File: physics/clm_lake.f90 [hunk 1]
```diff
@@ -498,14 +501,14 @@ SUBROUTINE clm_lake_run( &
...
-          if(lakedebug) then
+          if(debug_print) then
             write(0,*) 'Warning: wght2 is not 0..1: ',wght2
           endif
           wght2 = max(0.0_kind_lake,min(1.0_kind_lake,wght2))
         endif
         wght1 = 1.0_kind_lake - wght2
   
-        if(LAKEDEBUG .and. me==0) then
+        if(debug_print ) then
```
Comment (dustinswales) [inline physics/clm_lake.f90:504]: Do we want to do this on all processors?

**Review by dustinswales — APPROVED** [review]
Looks good.

**Review by Qingfu-Liu — APPROVED** [review]

File: physics/clm_lake.f90 (hunk 1 above)
Comment (SamuelTrahanNOAA) [reply physics/clm_lake.f90:504]: That lets us confirm the value is the same, and correct, on all ranks.

**Comment by zach1221** [comment]
@SamuelTrahanNOAA regression testing is finished on UFS-WM PR #1844. We can begin the merging process and start with this sub-component ccpp-physics PR.

**Comment by SamuelTrahanNOAA** [comment]
> We can begin the merging process and start with this sub-component ccpp-physics PR.

A fine idea. Unfortunately, I lack access to press the "merge" button. We shall have to wait for a code manager to resurface.

**Comment by zach1221** [comment]
Hello. @dustinswales @grantfirl would you please merge this PR?

**Comment by dustinswales** [comment]
@zach1221 Done. Sorry for the delay. 
FYI. I'm more of a backup code manager these days. @grantfirl @Qingfu-Liu @mkavulich are the mains, and I help out as needed. Grant is on holiday this week, so I'll make sure I'm watching PRs closer this week.

**Comment by zach1221** [comment]
> @zach1221 Done. Sorry for the delay. FYI. I'm more of a backup code manager these days. @grantfirl @Qingfu-Liu @mkavulich are the mains, and I help out as needed. Grant is on holiday this week, so I'll make sure I'm watching PRs closer this week.

@dustinswales Alright, understood. Thank you for helping!

**Comment by mkavulich** [comment]
@dustinswales Thanks for stepping in, sorry I missed the merge request here.

**Pull request merged**